File: src/main/java/hilfling/backend/hilfling/model/Gang.java
package hilfling.backend.hilfling.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Entity(name = "gangs")
@Table(name = "gangs")
public class Gang implements Serializable, BaseModel {
    public Gang() {
    }

    public Gang(String name) {
        this.name = name;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false, unique = true)
    private String name;

}

File: src/main/kotlin/hilfling/backend/hilfling/model/Gang.kt
package hilfling.backend.hilfing.model

import me.liuwj.ktorm.database.Database
import me.liuwj.ktorm.entity.Entity
import me.liuwj.ktorm.entity.sequenceOf
import me.liuwj.ktorm.schema.*

interface Gang : BaseModel<Gang> {
    companion object : Entity.Factory<Gang>()

    var name: String
}

object Gangs : BaseTable<Gang>("gang") {
    val name = varchar("name").bindTo { it.name }
}

val Database.gangs get() = this.sequenceOf(Gangs)